### PR TITLE
docs: Add a section to style on integer arithmetic

### PR DESCRIPTION
**Pull request by nagisa**
cc #8210

### CODE REVIEW

File: docs/practices/style.md [hunk 1]
```diff
@@ -257,6 +257,44 @@ external crates. The surprise factor for derivations sharing a name with the sta
...
+```
+
+This convention may be enforced by the `clippy::integer_arithmetic` lint.
+
+**Rationale:** By default the outcome of an overflowing computation in Rust depends on a few
+factors, most notably the compilation flags used. The quick explanation is that in debug mode the
+computations may panic if the result has overflowed, and when built with optimizations enabled,
+these computations will wrap-around instead.
+
+For nearcore and naerd we have opted to enable the panicking behaviour regardless of the
```
Comment (pugachAG) [inline docs/practices/style.md:288]: > naerd

typo

**Comment by pugachAG** [comment]
Maybe it would be worthwhile to mention that `x.checked_[add|sub|mul|div](y).expect("explanation why the operation is safe")` is a good default way of fixing `x [+|-|*|/] y` when the operation is guaranteed to be safe in the given context?

**Review by pugachAG — APPROVED** [review]
Thanks!

**Comment by pugachAG** [comment]
@akhi3030 could you please stamp this

**Review by akhi3030 — APPROVED** [review]

**Pull request merged**